go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -96,11 +96,11 @@ func (oc *DefaultNetworkController) newEgressFirewallRule(rawEgressFirewallRule
 		efr.to.nodeSelector = rawEgressFirewallRule.To.NodeSelector
 		efr.to.nodeAddrs = sets.NewString()
 		// validate node selector
-		_, err := metav1.LabelSelectorAsSelector(&rawEgressFirewallRule.To.NodeSelector)
+		_, err := metav1.LabelSelectorAsSelector(rawEgressFirewallRule.To.NodeSelector)
 		if err != nil {
 			return nil, fmt.Errorf("rule destination has invalid node selector, err: %v", err)
 		}
-		nodes, err := oc.watchFactory.GetNodesBySelector(rawEgressFirewallRule.To.NodeSelector)
+		nodes, err := oc.watchFactory.GetNodesBySelector(*rawEgressFirewallRule.To.NodeSelector)
```

Comment:
double checking here, this rawEgressFirewallRule.To.NodeSelector is now a pointer, there is no way this can be nil right (you will either have a matchLabel or a matchExpression set)? since we now do a * here just making sure we are safe...

Reply:
I think we should be good, if I do something like:
```
kind: EgressFirewall
apiVersion: k8s.ovn.org/v1
metadata:
  name: default
spec:
  egress:
  - to:
      nodeSelector:
    type: Allow
```
I get : `The EgressFirewall "default" is invalid: spec.egress[0].to: Invalid value: 0: spec.egress[0].to in body should have at least 1 properties` So guessing we can never really have a nil selector..

Reply:
Only way it seems to do it is not define nodeSelector at all in the yaml, but in that case we have min/max properties 1 so dns name or cidr will be defined, and in those cases we process either one of those before we ever attempt to parse node selector so we are covered.